appIcons.js

```diff
@@ -166,7 +166,13 @@ const MyAppIcon = new Lang.Class({
 
     _updateRunningStyle: function() {
         this.parent();
```

Comment:
Hi, sorry it took so long to answer.

1. That is correct in the case where we are using the windows counter (or the built-in theme option). If we are not using them, we still need to hide the "dots" (or bars, glow, etc...). Here is a video of what happens if we don't hide
them:
https://www.youtube.com/watch?v=1nuSk8IcenI

2. We need to tell the system that there was a change in the app list when we switch to another workspace. If we don't, then the list of running apps (non-favorites) is not refreshed properly. Here is a video of what happens: https://www.youtube.com/watch?v=E2vbn6vXd2k

3. When switching workspace, the focus is not refreshed unless we force it. This happens only for non-favorite apps.
Here is a video showing the problem:
https://www.youtube.com/watch?v=5o7s6VFAX20

I prepared all videos on Debian Testing (Gnome Shell 3.20). However, I tested the code (or variations of it, due to changes in the code!) on Gnome Shell 3.14, 3.20 and 3.18 to some extent.

Some of the fixes might not be super elegant, but from the use I had, not including them really feels buggy.

Thanks for looking into it!